### ISSUE TITLE
md5sum mismatch in resharded OPT-175B

### ISSUE THREAD

**Issue by dpfried** [facebookresearch/metaseq#590]
I've consolidated the 992 shards of the OPT-175B model into 8 files, following the instructions here: https://github.com/facebookresearch/metaseq/blob/main/projects/OPT/download_opt175b.md#reshard-the-shards 

but the md5sums for all of the shards don't match the ones in that README. Here are mine:

```
reshard-model_part-0.pt,ca52a4d1a3d1672b5249481169078469
reshard-model_part-1.pt,5c776fbc0f212df52fbf83b0ca4c7a64
reshard-model_part-2.pt,73e90bef8877f96674800712269149ae
reshard-model_part-3.pt,94d9442ec9fe04eab633df27d6e3ab5a
reshard-model_part-4.pt,dac742fe02fba6d500375bdd0a5113ea
reshard-model_part-5.pt,42485e9f21c759cae1f22079c0971a0b
reshard-model_part-6.pt,48ee0e983a8cde23b7544213e329afec
reshard-model_part-7.pt,ac62284b7f93d1f2a28eef6779cabeae
```

I've verified that the md5sums on all 992 individual shards that I've downloaded match the ones in https://github.com/facebookresearch/metaseq/blob/main/projects/OPT/assets/opt175b_md5sum_shards.csv .

One possible explanation is that I'm using CUDA 11.1 rather than 11.3 as specified on https://github.com/facebookresearch/metaseq/blob/main/docs/setup.md (we have a managed cluster that it's hard to install 11.3 on), but I didn't expect that to affect the model weights. Could it? I've otherwise followed the metaseq install directions exactly.

**Comment by suchenzang** [comment]
@tangbinh any chance the new reshard script would change the md5sums here?

**Comment by tangbinh** [comment]
> @tangbinh any chance the new reshard script would change the md5sums here?

Thank you for surfacing the issue @dpfried. @suchenzang Indeed, the new script `metaseq/scripts/reshard_fsdp.py` potentially changes the md5sums since previously we didn't reshard the optimizer state dicts as we do now. There's also an option to unflatten the weights. Let me update the docs to reflect these changes.

**Comment by tangbinh** [comment]
The issue has been addressed in #591. @dpfried Please let us know if you still have concern.

**Comment by guberti** [comment]
@tangbinh I don't think this issue has been resolved. When I use the latest version of the code (commit `b47f8d1`) and run the command specified in the `README`:
```bash
for j in {0..7}; do
    python -m metaseq.scripts.reshard_fsdp \
    --input-glob-pattern "/path/to/raw/checkpoints/checkpoint_last-model_part-$j-shard*.pt" \
    --output-shard-name "/path/to/resharded/checkpoints/reshard-model_part-$j.pt" \
    --num-output-shards 1 --skip-optimizer-state True --unflatten-weights True
done
```
I still get the following outputs:
```
ca52a4d1a3d1672b5249481169078469	reshard-model_part-0.pt
5c776fbc0f212df52fbf83b0ca4c7a64	reshard-model_part-1.pt
73e90bef8877f96674800712269149ae	reshard-model_part-2.pt
94d9442ec9fe04eab633df27d6e3ab5a	reshard-model_part-3.pt
dac742fe02fba6d500375bdd0a5113ea	reshard-model_part-4.pt
42485e9f21c759cae1f22079c0971a0b	reshard-model_part-5.pt
48ee0e983a8cde23b7544213e329afec	reshard-model_part-6.pt
ac62284b7f93d1f2a28eef6779cabeae	reshard-model_part-7.pt
```
